ktor-client/ktor-client-plugins/ktor-client-logging/common/src/io/ktor/client/plugins/logging/Logging.kt

```diff
@@ -191,6 +191,7 @@ public val Logging: ClientPlugin<LoggingConfig> = createClientPlugin("Logging",
         contentLength: Long?,
         headers: Headers,
         method: HttpMethod,
+        logLines: MutableList<String>,
```

Comment:
I would pass `log: (String) -> Unit` instead, cause you're not using any list functions except add

Reply:
We also can make this function top-level internal and have some unit tests

Reply:
> I would pass `log: (String) -> Unit` instead, cause you're not using any list functions except add

What's the point of that?

> We also can make this function top-level internal and have some unit tests

What this function would do? How would we test it?

Reply:
> What's the point of that?
You don't allow future modification to drop the content of the list

> What this function would do? How would we test it?
All it is doing atm. I just thought we might replace some flaky integration tests with unit tests on this function

Reply:
> You don't allow future modification to drop the content of the list

I don't see a problem in that if the decision is thoughtful.

> All it is doing atm. I just thought we might replace some flaky integration tests with unit tests on this function.

There would be a race condition If the client made multiple simultaneous requests while using the same buffer for log lines.

Reply:
Also, my integration tests aren't flaky because I use the `testApplication`.